pkg/common/utils.go

```diff
@@ -0,0 +1,22 @@
+package common
+
+import (
+	"math"
+	"math/rand"
+	"time"
+)
+
+// ExponentialBackoff is an algorithm which is used to spread out repeated execution of task (usually reconnection or packet
+// sending task to avoid network congestion or to decrease server load). In exponential backoff, wait time is increased
+// exponentially till threshold value. We have introduced jitter here to avoid thundering herd problem in future.
+func ExponentialBackoff(task func() error, baseTimeInMillis float64, multiplier float64, thresholdInMillis float64, retries int) error {
+	err := task()
+	backoffTime := math.Min(baseTimeInMillis, thresholdInMillis)
+	for i := 0; err != nil && i < retries; i++ {
+		sleepTime := rand.Float64() * backoffTime
```

Comment:
This can reduce the backoffTime as compared to previous retry. Jitter can be introduced by taking a fraction of the multiplier instead.

Reply:
changed it to use existing implementation.